apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -395,23 +403,58 @@ export const submitChatContents = createAsyncThunk(
     }
 
     thunkAPI.dispatch(clearChatMessagePending());
-    chatApiResponse.messages.forEach(message =>
-      thunkAPI.dispatch(addChatMessage({...message, timestamp: Date.now()}))
-    );
+    chatApiResponse.messages.forEach(message => {
+      thunkAPI.dispatch(addChatEvent({...message, timestamp: Date.now()}));
+    });
+  }
+);
+
+// This thunk adds a chat event to chatEventsCurrent (displayed in current chat workspace)
+// if hideForParticipants != true and then logs the event to the backend for all chat events
+// except notifications with includeInHistory != true.
+export const addChatEvent = createAsyncThunk(
```

Comment:
Oh one quick thing (sorry I didn't notice this before) - does this need to be an async thunk? Is there any async work  happening in this function?

Reply:
Good catch! Nope, the async stuff is now in `ChatEventLogger`. Updated!